docs/components/HookTypeScriptHints.tsx

```diff
@@ -1,32 +1,22 @@
 export { HookTypeScriptHints }
 
 import React from 'react'
-import { assert, Link } from '@brillout/docpress'
+import { assert, Link, Warning } from '@brillout/docpress'
 
-function HookTypeScriptHints({ hookTypeName, wrongDefinition }: { hookTypeName: string; wrongDefinition: string }) {
-  assert(hookTypeName)
-  assert(wrongDefinition)
+function HookTypeScriptHints({ hookTypeName }: { hookTypeName: `${string}Sync` | `${string}Async` }) {
+  assert(hookTypeName.endsWith('Sync') || hookTypeName.endsWith('Async'))
+  const isSync = hookTypeName.endsWith('Sync')
   return (
     <>
-      <blockquote>
-        <p>
-          Do not omit the <code>ReturnType&lt;{hookTypeName}&gt;</code> annotation (i.e. do not write{' '}
-          <code>{wrongDefinition}</code>), otherwise TypeScript won't strictly check the return type for unknown extra
-          properties. See also:
-          <ul>
-            <li>
-              <a href="https://stackoverflow.com/questions/63006497/typescript-why-not-strict-return-type">
-                Stack Overflow &gt; Typescript: why not strict return type?
-              </a>
-            </li>
-            <li>
-              <a href="https://stackoverflow.com/questions/57214325/typescript-excess-properties-are-not-checked-in-lambdas-without-explicit-return">
-                Stack Overflow &gt; TypeScript: Excess properties are not checked in lambdas without explicit return
-              </a>
-            </li>
-          </ul>
-        </p>
-      </blockquote>
+      <Warning>
+        Don't omit <code>ReturnType&lt;{hookTypeName}&gt;</code> (don't write{' '}
+        <code>{`const onRenderHtml: ${hookTypeName} = ${isSync ? '' : 'async '}(pageContext) => {`}</code>), otherwise
```

Comment:
`onRenderHtml` will be wrong here most of the time

Reply:
Already fixed :) thanks!